cmd/find-main.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Minio Client (C) 2014, 2015 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+)
+
+// find specific flags
+var (
+	findFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "name",
+			Usage: "Find files of a matching name",
+		},
+		cli.StringFlag{
+			Name:  "path",
+			Usage: "Find global matches on the entire path",
+		},
+		cli.StringFlag{
+			Name:  "regex",
+			Usage: "Matches all files with PCRE regular expression",
+		},
+		cli.StringFlag{
+			Name:  "print",
+			Usage: "Prints the found files given args to STDOUT",
+		},
+		cli.StringFlag{
+			Name:  "mcexec",
```

Comment:
> If the user has their own command in the file path which they are trying to call (like a shell script).

User has it won't work since it doesn't understand the abstraction which `mc` provides between filesystem and object storage. 

> To accomplish this notion of "subsequent" has to first be defined, and does this vary from os to os. Ex, on Windows () can be used to group or nest commands (https://ss64.com/nt/syntax-brackets.html), but on Unix the usage of () can allow the output of a command to replace itself (http://www.faqs.org/docs/bashman/bashref_30.html).

No this is verbatim there is no OS level interpretation here, you are not even passing this to bash or windows command line for it to be interpreted. The interpretation only happens after you constructed the right command line.